src/interactionManager/interaction.ts

```diff
@@ -417,15 +404,17 @@ export class Interaction<F extends Flow<any> = Flow<any>> extends Transportable
    * Validate an interaction token and process it to update the interaction
    * state (via the associated {@link Flow})
    *
-   * @param token - the token to
+   * @param token - the token to process
+   * @param storeToken - whether to commit the token to storage (default: true)
    * @returns Promise<boolean> whether or not processing was successful
-   * @throws AppError<InvalidToken> with `origError` set to the original token
+   * @throws SDKError<InvalidToken> with `origError` set to the original token
    *                                validation error from the jolocom library
    * @category Basic
    *
    */
   public async processInteractionToken<T>(
     token: JSONWebToken<T>,
+    storeToken = true
```

Comment:
This was not added for that reason. This was added to allow for recreating an Interaction by processing tokens loaded from storage, which of course should not be committed back to storage (they would get saved twice basically).
I actually don't like this either, but I think the proper fix is to not touch storage **at all** on this layer (i.e. commit tokens to storage on a different SDK layer), but that is going to lead to a bunch of breaking changes. 

---

The change that was actually introduced to allow for half-assed multitenancy is the change in `agent.processJWT` which allows it to skip processing a token that was already seen before, if it is the last one seen for that interaction. I actually think that change might be useful for other reasons, and maybe the comment about it can be removed (it can stay even after proper multitenancy support is added)

Reply:
Also please note that adding proper multitenancy support is most probably going to lead to a whole bunch of breaking changes, I don't see it coming before a 2.0

Reply:
it's possible though to check if the token is already stored though, no? then a flag is not necessary

Reply:
Well yes that is possible of course. Not great for performance, but I suppose that it will be removed later anyway.

Reply:
it's ugly af >_>